core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -573,35 +562,41 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
                               conditionalPath,
                               Vector(signer.publicKey))
 
-    ECSignatureParams(inputInfo, signer, hashType)
+    ECSignatureParams(inputInfo, txVec.head.transactionSpent, signer, hashType)
   }
 
   private def changeToWitnessUTXO(
       transactionOutput: TransactionOutput): InputPSBTMap = {
     val newElements = transactionOutput.scriptPubKey match {
-      case _: WitnessScriptPubKey =>
+      case rs: WitnessScriptPubKey if !rs.isInstanceOf[WitnessScriptPubKeyV0] =>
         filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
```

Comment:
This exhaustive-check-breaking `if` clause can be avoided by simply moving this case beneath the case that checks for `WitnessScriptPubKeyV0`